vowpalwabbit/global_data.h

```diff
@@ -344,10 +345,24 @@ enum label_type_t
   cs,       // cost-sensitive
   multi,
   mc,
-  ccb       // conditional contextual-bandit
+  ccb  // conditional contextual-bandit
 };
 }
 
+class rand_state
+{
+ private:
+  uint64_t random_state;
+
+ public:
+  constexpr rand_state() : random_state(0) {}
+  rand_state(uint64_t initial) : random_state(initial) {}
+  constexpr uint64_t get_current_state() const noexcept { return random_state; }
+  float get_and_update_random() { return merand48(random_state); }
+  float get_random() const { return merand48_noadvance(random_state); }
```

Comment:
Why not call this merand48_noadvance?

Reply:
merand48/merand48_noadvance are pretty arbitrary names which don't hold much meaning. These names can hide the underlying operations while providing a descriptive interface for what the function does